section/algorithms.html

```diff
@@ -1,539 +1,686 @@
+          </li>
+          <li>
+            Perform a [=fetch request=] with |request|, given a [=response=]:
+            <ol>
+              <li>
+                If {{"NetworkError"}} is thrown, notify the user that the browser could not cancel the grant.
+              </li>
+              <li>
+                If the [=status=] is `204`, the grant was successfully canceled and the storage can be cleared.
+                <p class="note">
+                  When clearing the storage all user data should be removed. The only properties
+                  that should not be cleared are: {{Storage/privateKey}}, {{Storage/publicKey}}, {{Storage/kid}}.
+                </p>
+              </li>
+            </ol>
+          </li>
+        </ol>
+      </div>
+    </section>
+  </section>
+
   <section>
-  <h3>Incoming payment grant request</h3>
-    <p>The incoming payment grant request refers to the <a data-cite="opas/operations/post-request#">Open Payments grant request</a>.
-      This request is non-interactive: implementations MUST NOT include an {{GrantRequest.interact}} and
-      the authorization server will return a {{Grant}} directly if successful.</p>
-    <div class="algorithm">
-      When asked to <dfn>send an incoming payment grant request</dfn>, given |websiteWalletAddressDetails:WalletAddressDetails|
-      perform the following steps.
-    <ol>
-        <li>
-          Let |userWalletDetails:WalletAddressDetails?| be the result of [=get user wallet=].
-        </li>
-        <li>If |userWalletDetails| is null, then return failure.</li>
-      <li>
-        Let |authServer:URL| be the result of running [=URL parser=] with |websiteWalletAddressDetails|'s {{WalletAddressDetails/authServer}} property.
-      </li>
-      <li>
-        If |authServer| [=url/scheme=] is not `https`, return failure.
-      </li>
-      <li>
-        <p>Construct a {{ GrantRequest }} |grantRequest: GrantRequest| as follows:</p>
+    <h3>Resource server operations (RS)</h3>
+    <section>
+      <h4>Create incoming payment</h4>
+      <p>The create incoming payment request refers
+        to the <a data-cite="oprs/operations/create-incoming-payment#">Open Payments create incoming payment</a>.</p>
+      <div class="algorithm">
+        When asked to <dfn>send a create incoming payment request</dfn>,
+        given |websiteWalletAddressDetails:WalletAddressDetails|, |amountValue:DOMString|,
+        |incomingPaymentGrant:Grant|, perform the following steps.
         <ol>
           <li>
-            Set |grantRequest|'s {{GrantRequest/client}} to |userWalletDetails|.{{WalletAddressDetails/id}}.
+            Let |resourceServer:URL| be the result of running [=URL parser=] with |websiteWalletAddressDetails|'s {{WalletAddressDetails/resourceServer}} property.
+          </li>
+          <li>
+            If |resourceServer| [=url/scheme=] is not `https`, return failure.
           </li>
```

Comment:
I wonder if we should remove these steps from each algorithm, and instead validate these when we fetch the wallet address details. That way, the `WalletAddressDetails` will only contain valid URLs/data.